src/API/Google/Ads.php

```diff
@@ -323,4 +324,77 @@ private function get_merchant_link( int $merchant_id ): MerchantCenterLink {
 
 		throw new Exception( __( 'Merchant link is not available to accept', 'google-listings-and-ads' ) );
 	}
+
+	/**
+	 * Check if the user has accepted the customer data terms for enhanced conversion tracking.
+	 * Returns false for any account that fails.
+	 *
+	 * @return boolean
+	 */
+	public function get_accepted_customer_data_terms(): ?bool {
+		$ads_id = $this->options->get_ads_id();
+
+		try {
+			$accepted_terms = $this->options->get( OptionsInterface::CUSTOMER_DATA_TERMS, null );
+
+			// Retrieve the terms acceptance data from options.
+			if ( null !== $accepted_terms ) {
+				return apply_filters( 'woocommerce_gla_ads_enhanced_conversion_customer_data_terms', boolval( $accepted_terms ) );
+			}
+
+			$customer = ( new AdsAccountQuery() )
+				->set_client( $this->client, $ads_id )
+				->columns( [ 'customer.conversion_tracking_setting.accepted_customer_data_terms' ] )
+				->get_result()
+				->getCustomer();
+
+			if ( ! $customer ) {
+				return false;
+			}
+
+			$conversion_tracking_setting = $customer->getConversionTrackingSetting();
+
+			$accepted = $conversion_tracking_setting->getAcceptedCustomerDataTerms();
+
+			// Save the data terms in options as those cannot be reverted.
+			$this->options->update( OptionsInterface::CUSTOMER_DATA_TERMS, $accepted );
+
+			return apply_filters( 'woocommerce_gla_ads_enhanced_conversion_customer_data_terms', $accepted );
+		} catch ( ApiException $e ) {
+			do_action( 'woocommerce_gla_ads_client_exception', $e, __METHOD__ );
+		}
+
+		return null;
+	}
+
+	/**
+	 * Updates the enhanced ads conversion status.
+	 *
+	 * @param string $status The status value
+	 *
+	 * @return string
+	 * @throws Exception When the status is invalid.
+	 */
+	public function update_enhanced_conversion_status( string $status ): string {
+		$status = strtolower( $status );
+		// Ensure that the option belongs to one of the predefined values.
+		if ( ! in_array( $status, [ 'enabled', 'disabled', 'pending' ], true ) ) {
+			throw new Exception( __( 'Invalid state for enhanced conversion', 'google-listings-and-ads' ), 400 );
+		}
+
+		$this->options->update( OptionsInterface::ENHANCED_CONVERSION_STATUS, $status );
+
+		return $status;
+	}
+
+	/**
+	 * Retrieve the enhanced ads conversion status. Possible values are: enabled, disabled and pending
+	 *
+	 * @return string|null
+	 */
+	public function get_enhanced_conversion_status(): ?string {
```

Comment:
This function has a strict return type, but getting an option doesn't enforce the type since it gets the option directly from the DB which calls `maybe_unserialize`. It shouldn't happen unless DB data is garbled, but is that something we want to protect against? Seems right now the fatal error will bubble up and show in the UI, since it didn't get a valid status it will assume it still needs to enable the conversion status so it's not blocking. But I'm wondering if this should be caught earlier as the UI can't really do anything with this error.

Reply:
Thank you @mikkamp 

We are only updating the `OptionsInterface::ENHANCED_CONVERSION_STATUS` option in `update_enhanced_conversion_status` method in same file. And since it is being called from the API (for which we have added the above mentioned schema), there should be very less probability of the data being any other than string.

However; I have added an extra check if the data is not scalar then we return `null`, if it is scalar, we return the `strval` of the result.